lib/train/extras/os_detect_arista_eos.rb

```diff
@@ -0,0 +1,30 @@
+# encoding: utf-8
+# author: Jere Julian
+#
+# Arista EOS has 2 modes. Most compliance tests will use the network CLI
+# but when working with vagrant, its common to encounter the raw bash shell.
+require 'json'
+
+module Train::Extras
+  module DetectAristaEos
+    def detect_arista_eos
+      if unix_file?('/usr/bin/FastCli')
+        output = @backend.run_command('FastCli -p 15 -c "show version | json"').stdout
+        @platform[:name] = 'arista_eos_bash'
+        family = 'fedora'
+      else
+        output = @backend.run_command('show version | json').stdout
+      end
+
+      unless output.empty?
+        eos_ver = JSON.parse(output)
+        @platform[:name] = @platform[:name] || 'arista_eos'
+        family ||= 'arista_eos'
```

Comment:
I was thinking about this a bit.   If we always mark EOS as family: arista_eos and include that as "linux", then resources will fail instead of skipping when we're in the usual CLI mode unless from the os detection I can specify to default to prefixing all linux commands with `bash` to drop to a linux shell (Ex: `bash uname -a` or `bash sudo cat /etc/passwd`).   

Alternatively, I could just patch up the resources as we make use of them.